crd-generator/maven-plugin/pom.xml

```diff
@@ -0,0 +1,157 @@
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-plugin-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>generate-descriptor</id>
+            <goals>
+              <goal>descriptor</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>generated-helpmojo</id>
+            <goals>
+              <goal>helpmojo</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-invoker-plugin</artifactId>
+        <configuration>
+          <cloneProjectsTo>${project.build.directory}/it</cloneProjectsTo>
+          <cloneClean>true</cloneClean>
+          <settingsFile>src/it/settings.xml</settingsFile>
+          <postBuildHookScript>verify</postBuildHookScript>
+          <addTestClassPath>true</addTestClassPath>
+          <skipInvocation>${skipTests}</skipInvocation>
+          <streamLogs>true</streamLogs>
+          <invokerPropertiesFile>invoker.properties</invokerPropertiesFile>
+          <scriptVariables>
+            <version>${project.version}</version>
+            <artifactId>${project.artifactId}</artifactId>
+            <groupId>${project.groupId}</groupId>
+          </scriptVariables>
+          <setupIncludes>
+            <setupInclude>test-dependencies/pom.xml</setupInclude>
+          </setupIncludes>
+        </configuration>
+        <executions>
+          <execution>
+            <id>integration-tests</id>
+            <goals>
+              <goal>install</goal>
+              <goal>run</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>com.diffplug.spotless</groupId>
+        <artifactId>spotless-maven-plugin</artifactId>
+        <version>${spotless-maven-plugin.version}</version>
+        <configuration>
+          <java>
+            <includes>
+              <include>src/main/java/**/*.java</include>
+              <include>src/test/java/**/*.java</include>
+              <include>src/it/**/src/main/java/**/*.java</include>
+            </includes>
+          </java>
+          <pom>
+            <includes>src/it/**/pom.xml</includes>
+            <sortPom/>
+          </pom>
+          <groovy>
+            <includes>
+              <include>src/it/**/*.groovy</include>
+            </includes>
+            <importOrder/>
+            <greclipse/>
+          </groovy>
+        </configuration>
+      </plugin>
```

Comment:
Why is this needed?
isn't the plugin configuration in the parent pom enough?
can these changes be part of the parent pom instead so that we keep the plugin configuration in a single place?

Reply:
It's required if the formatter should consider groovy files, too.
This isn't configured global and I was unsure if it should be configured global because of the impact.
If we don't care about the groovy files we can remove the whole plugin config and fallback to the defaults.

Is groovy file formatting important for you, so that we should keep it either here or global? Or should I remove it?

Reply:
I understand this is for verify.groovy and other helper scripts (which are also present in other modules).
I don't mind one or the other, but I do think we need to centralize configuration.
If adding this globally implies making changes in other modules due to formatting, then remove (in the end these files aren't very important and that many).
If not, then please add the configuration globally.

Reply:
I have moved the groovy / integration test spotless config to the parent pom.
First with a minimal config required to cover only the new integration test. The second commit increases the scope and includes corresponding formatting fixes. 
If it's too much, I can revert the second commit.

But I discovered a small "problem": It seems like we are affected by bug of the spotless groovy formatter / greclipse. If formatting is checked or applied with the maven plugin an empty temp directory will be created in the project root.

https://github.com/diffplug/spotless/issues/2019

Can we deal with that or do we need a workaround?

Reply:
I guess we can try and live with this for now. If it gets annoying we can disable.
For the second commit, I'd need to check first.